components/connect/connect.c

```diff
@@ -389,6 +390,45 @@ void wifi_init(void * pvParameters)
             ESP_LOGI(TAG, "ESP_WIFI setting hostname to: %s", hostname);
         }
 
+        /* Initialize mDNS */
+        err = mdns_init();
+        if (err != ESP_OK) {
+            ESP_LOGW(TAG, "mDNS/Avahi initialization failed: %s", esp_err_to_name(err));
+            ESP_LOGW(TAG, "Device will not be discoverable via mDNS/Bonjour/Avahi");
+        } else {
+            ESP_LOGI(TAG, "mDNS/Avahi initialized successfully - device discoverable on network");
+
+            /* Set mDNS hostname */
+            err = mdns_hostname_set(hostname);
+            if (err != ESP_OK) {
+                ESP_LOGW(TAG, "mDNS hostname setup failed: %s", esp_err_to_name(err));
+                ESP_LOGW(TAG, "Device hostname not set for mDNS discovery");
+            } else {
+                ESP_LOGI(TAG, "mDNS hostname set to: %s.local", hostname);
+                ESP_LOGI(TAG, "Access device at: http://%s.local", hostname);
+            }
+
+            /* Set mDNS instance name */
+            err = mdns_instance_name_set("ESP-Miner Web Server");
```

Comment:
Is this instance name exposed somewhere, or is it internal only? If exposed, maybe we can give it a bit nicer name. Maybe device model, f.e. `Bitaxe Gamma` from DeviceConfig?